engine/table/src/main/java/io/deephaven/engine/util/WindowCheck.java

```diff
@@ -265,89 +333,366 @@ protected void process() {
                     downstream.added = RowSetFactory.empty();
                     downstream.removed = RowSetFactory.empty();
                     downstream.shifted = RowSetShiftData.EMPTY;
-                    downstream.modifiedColumnSet = reusableModifiedColumnSet;
+                    downstream.modifiedColumnSet = result.getModifiedColumnSetForUpdates();
                     downstream.modifiedColumnSet().clear();
-                    downstream.modifiedColumnSet().setAll(mcsNewColumns);
+                    downstream.modifiedColumnSet().setAll(mcsResultWindowColumn);
```

Comment:
Why did you decide to use the methods for `modifiedColumnSet` one line after just directly setting it?